internal/attestation/crafter/crafter.go

```diff
@@ -487,6 +511,12 @@ func (c *Crafter) AddMaterial(ctx context.Context, attestationID, key, value str
 	}
 
 	// 3 - Craft resulting material
+	return c.addMaterial(ctx, m, attestationID, key, value, casBackend, runtimeAnnotations)
+}
+
+// addMaterials adds the incoming material m to the crafting state
+func (c *Crafter) addMaterial(ctx context.Context, m *schemaapi.CraftingSchema_Material, attestationID, key, value string, casBackend *casclient.CASBackend, runtimeAnnotations map[string]string) error {
```

Comment:
You might don't need to pass the key since `m` has a `name` correct?

Reply:
Hmm yes, it's included in `m` we can use it.